### ISSUE TITLE
Merge rule-non-nested-empty-line-before and rule-nested-empty-line-before

### ISSUE THREAD

**Issue by rontav** [stylelint/stylelint#2229]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug

> Which rule, if any, is this issue related to?

rule-nested-empty-line-before with except "after-rule"

> What CSS is needed to reproduce this issue?

```css
a {
  b {
    color:  red;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "rule-nested-empty-line-before": [
      "always",
      {
        "except": [
          "after-rule"
        ]
      }
    ],
  }
}
```

> Which version of stylelint are you using?

7.7.1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

VSCode plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, SCSS nesting

> What did you expect to happen?

No errors to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
2:2    Expected empty line before nested rule (rule-nested-empty-line-before)
```

**Comment by rontav** [comment]
Can someone look at [this commit](https://github.com/rontav/stylelint/commit/1dc6a048646bb07e23028af54a180e27d601a128)? 
I think in this case, media rules should be considered rules because the resulted code is more consistent.

Example:
```css
a {
  color: green;

  @media {
    color: red;
  }
  b {
    color: blue;
  }
}
```
vs
```css
a {
  color: green;
  @media {
    color: red;
  }

  b {
    color: blue;
  }
}
```

**Comment by rontav** [comment]
I'm ok with except after-at-rule. In the end, I want to be able to config stylelint to enforce my coding style in my projects.

Yes, that's the coding style I want to enforce. With one observation: I usually put media rules at the begining (before f) or at the end (after h). Should I open a new issue for this? :smile:

**Comment by jeddy3** [comment]
> I usually put media rules at the begining (before f) or at the end (after h). Should I open a new issue for this?

Working through this has been very helpful as it has uncovered another inconsistency. Currently, we have:

- the rule `at-rule-empty-line-before` with the option `"inside-block"`.
- the rules `rule-non-nested-empty-line-before` and `rule-nested-empty-line-before`.

That's two different approaches to solving the same problem: a single rule with `"inside-*"` options vs two separated nested and non nested rules without `"inside-*"` options.

@stylelint/core I think we should use the single rule with `"inside-*"` options approach for the rule constructs because:

1. We already use the `"inside-*"` [options](https://github.com/stylelint/stylelint/blob/master/lib/rules/color-named/README.md#ignore-inside-function) [else](https://github.com/stylelint/stylelint/blob/master/lib/rules/indentation/README.md#inside-parens) [where](https://github.com/stylelint/stylelint/blob/master/lib/rules/custom-property-empty-line-before/README.md#inside-single-line-block) and it seems to be working well.
2. I think the configuration would end up being more consistent with how _all_ the `except` and `ignore` options work (and therefore more intuitive) i.e. target a _construct_ (e.g. at-rule, rule, comment, declaration etc), set a primary option, override that primary option for particular patterns using the `ignore` and `except` secondary option. For example...

This code style would be enforced by:

```css
/* comment as first name */
@media {}

a {
  b {
    color: red;
  }
}

c {
  --custom-prop: 10px;

  color: red;
  top: 5px;

  d {}
  /* comment */
  e {}
  f {}
  @media {}
}

g {
  @media {}
}

```

```js
{
  "rules": {
    "at-rule-empty-line-before": [
      "always",
      {
        "except": [
          "after-comment",
          "inside-block-and-after-opening-brace",
          "inside-block-and-after-rule"
        ]
      }
    ],
    "custom-property-empty-line-before": [
      "always",
      {
        "except": [
          "after-comment",
          "inside-block-and-after-opening-brace",
        ]
      }
    ],
    "declaration-empty-line-before": [
      "always",
      {
        "except": [
          "after-comment",
          "inside-block-and-after-opening-brace",
        ]
      }
    ],
    "rule-empty-line-before": [ // new consistent combined rule with inside-* options
      "always",
      {
        "except": [
          "after-comment",
          "inside-block-and-after-opening-brace",
          "inside-block-and-after-at-rule",      
          "inside-block-and-after-rule"
        ]
      }
    ],
  }
}
```

Thoughts?

@rontav FYI, use the [`stylelint-order`](https://github.com/hudochenkov/stylelint-order) plugin pack for all your ordering needs.  You can use [`order/declaration-block-order`](https://github.com/hudochenkov/stylelint-order/blob/master/rules/declaration-block-order/README.md) to enforce media at-rules before _or_ after rules. Pick one style per code base though.
